src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_BinaryOperator.cs

```diff
@@ -120,6 +127,13 @@ public BoundExpression VisitBinaryOperator(BoundBinaryOperator node, BoundUnaryO
 
             for (BoundBinaryOperator? current = node; current != null && current.ConstantValue == null; current = current.Left as BoundBinaryOperator)
             {
+                // The regular visit mechanism will handle this.
+                if (current.InterpolatedStringHandlerData is not null)
```

Comment:
Is it possible to encounter a binary op node with `InterpolatedStringHandlerData` aside from the top node? I couldn't find any test in `InterpolationTests` affected by this `if` block.

Reply:
I can add a test for this, but yes it's possible. Consider a series of strings added to an object: the strings will be combined with the default builder, then the standard string + object code will be run.